### PR TITLE
feat(scene): enable scene backgrounds

### PR DESCRIPTION
**Pull request by haweston**
## Overview
Update layout of fog settings to not require expander click and toggle color picker/distance inputs based on enable checkbox

Add scene background settings with color picker.  Does not support background textures.

https://github.com/awslabs/iot-app-kit/assets/109186219/533ea9f7-b88f-4986-9e0b-cdda28c39be4

## Verifying Changes
Use storybook to open a scene composer
visit settings
go to scene settings
use the enable fog and enable background to test setting and see the colors change in the 3d scene

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/interfaces/interfaces.tsx [hunk 1]
```diff
@@ -148,3 +149,7 @@ export interface IFogSettings {
   near: number;
   far: number;
 }
+
+export interface ISceneBackgroundSetting {
+  colorHex?: number;
```
Comment (sheilaXu) [inline packages/scene-composer/src/interfaces/interfaces.tsx:154]: do we need to store the hex number or we can just store the hex string? looks like the color picker returns the string, and the threejs color can handle color string. and we will need to do some extra conversion between color number and string. can we just save the color hex string then?

I found many instances in our code that we allow color to be either number or string, then we add many conversion steps, but the actual value we get from the picker and set to the 3D component can just be a color string generically. then we can simplify some conversion code.

Same comment for the `colorHex` in `IFogSettings`

File: packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx [hunk 2]
```diff
@@ -94,54 +93,53 @@ export const FogSettingsEditor: React.FC = () => {
...
+        </Checkbox>
+        {!!fogSettings && (
+          <>
+            <ColorSelectorCombo
+              color={`#${internalColor.toString(16).padStart(6, '0')}`}
+              onSelectColor={(pickedColor) => onColorChange(pickedColor)}
+              onUpdateCustomColors={(chosenCustomColors) =>
+                setTagColorsSceneProperty(KnownSceneProperty.TagCustomColors, chosenCustomColors)
+              }
+              customColors={tagStyleColors}
```
Comment (sheilaXu) [inline packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx:112]: is this intentional to update `TagCustomColors` when changing Fog color?

File: packages/scene-composer/src/components/panels/scene-settings/SceneBackgroundSettingsEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          {intl.formatMessage({ defaultMessage: 'Enable Background', description: 'checkbox label' })}
+        </Checkbox>
+        {!!backgroundSettings && (
+          <ColorSelectorCombo
+            color={`#${internalColor.toString(16).padStart(6, '0')}`}
+            onSelectColor={(pickedColor) => onColorChange(pickedColor)}
+            onUpdateCustomColors={(chosenCustomColors) =>
+              setTagColorsSceneProperty(KnownSceneProperty.TagCustomColors, chosenCustomColors)
+            }
+            customColors={tagStyleColors}
```
Comment (sheilaXu) [inline packages/scene-composer/src/components/panels/scene-settings/SceneBackgroundSettingsEditor.tsx:77]: same here for whether scene background color should share the same custom colors list with tag

File: packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx (hunk 2 above)
Comment (haweston) [reply packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx:112]: We don't have fog or background color lists from UX yet so this was a stop gap for the PoC to get feedback.  But I think we can make new fog and background custom colors lists now and just have them initialized to the same values as our tag colors but save custom colors off separately from tags.  Then when we get colors from UX we can just change the initialization and not have to change each editor as well.

File: packages/scene-composer/src/interfaces/interfaces.tsx (hunk 1 above)
Comment (haweston) [reply packages/scene-composer/src/interfaces/interfaces.tsx:154]: Let me give this ago, as I agree that the string to number conversion is annoying. I think when I looked at the three.js docs only saw hex values as numbers and strings as 'blue' etc. based on the examples but the full spec makes it look like it also takes strings of format '#ff00ff'

File: packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx (hunk 2 above)
Comment (sheilaXu) [reply packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx:112]: the tag custom colors are a list of custom colors the user previously has selected. it is also initialized with an empty list

Comment (haweston) [reply packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx:112]: You're right, I was confusing it with the ColorSelectorCombos initial list of colors of non-custom colors which is set in TagColors.ts. But I'm breaking them up to all use their own custom save list even if the ColorSelectorCombo is still initialized from TagColors for non-custom colors.

File: packages/scene-composer/src/interfaces/interfaces.tsx (hunk 1 above)
Comment (haweston) [reply packages/scene-composer/src/interfaces/interfaces.tsx]: Swapped fog & scene background to use color: string

File: packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx (hunk 2 above)
Comment (haweston) [reply packages/scene-composer/src/components/panels/scene-settings/FogSettingsEditor.tsx]: added fogCustom and backgroundCustom save system separate from tagCustom

File: packages/scene-composer/src/components/panels/scene-settings/SceneBackgroundSettingsEditor.tsx (hunk 3 above)
Comment (haweston) [reply packages/scene-composer/src/components/panels/scene-settings/SceneBackgroundSettingsEditor.tsx]: see above, added backgroundCustom save separate from tagCustom

**Review by sheilaXu — APPROVED** [review]

**Pull request merged**